### PR TITLE
Fix not emitted statement in then clauses producing syntactically invalid output

### PR DESCRIPTION
**Pull request by weswigham**
Fixes one of the errors discovered by #32009.

### CODE REVIEW

File: src/compiler/factory.ts [hunk 1]
```diff
@@ -1544,8 +1544,8 @@ namespace ts {
     export function createIf(expression: Expression, thenStatement: Statement, elseStatement?: Statement) {
         const node = <IfStatement>createSynthesizedNode(SyntaxKind.IfStatement);
         node.expression = expression;
-        node.thenStatement = thenStatement;
-        node.elseStatement = elseStatement;
+        node.thenStatement = isNotEmittedOrPartiallyEmittedNode(thenStatement) ? setTextRange(setOriginalNode(createEmptyStatement(), thenStatement), thenStatement) : thenStatement;
```
Comment (rbuckton) [inline src/compiler/factory.ts]: Shouldn't we do this for any "embedded statement" as well (i.e. `for`, `do`, `while`, labeled statement, etc.)

Comment (weswigham) [reply src/compiler/factory.ts]: Hm, probably. Should I write up some more test cases and generalize a helper, `asEmbeddedStatement` that is this ternary?

Comment (rbuckton) [reply src/compiler/factory.ts]: Yeah, but you probably just want `isNotEmittedStatement`. A `PartiallyEmittedExpression` is an "expression" and wouldn't be legal in this context (it basically just wraps an expression to add comments/sourcemaps from a parent node that was elided, like eliding parens in a cast `(x as foo)`).

**Comment by weswigham** [comment]
@rbuckton 👍 👎 ?

**Comment by weswigham** [comment]
Imma merge this since it seems like an improvement and I've integrated your review comment ❤️

**Pull request merged**